cmd/oras/internal/option/pusher.go

```diff
@@ -18,26 +18,36 @@ package option
 import (
 	"context"
 	"encoding/json"
+	"errors"
+	"fmt"
 	"os"
+	"regexp"
+	"strings"
 
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 	"github.com/spf13/pflag"
 	"oras.land/oras-go/v2/content"
 )
 
+const (
+	annotationManifest = "$manifest"
+)
+
 // Pusher option struct.
 type Pusher struct {
-	ManifestExportPath     string
-	PathValidationDisabled bool
-	ManifestAnnotations    string
+	ManifestExportPath      string
+	PathValidationDisabled  bool
+	ManifestAnnotations     string
+	ManifestAnnotationSlice []string
 
 	FileRefs []string
 }
 
 // ApplyFlags applies flags to a command flag set.
 func (opts *Pusher) ApplyFlags(fs *pflag.FlagSet) {
 	fs.StringVarP(&opts.ManifestExportPath, "export-manifest", "", "", "export the pushed manifest")
-	fs.StringVarP(&opts.ManifestAnnotations, "manifest-annotations", "", "", "manifest annotation file")
+	fs.StringArrayVarP(&opts.ManifestAnnotationSlice, "annotation", "a", []string{}, "manifest annotations")
```

Comment:
Why not default to nil

Reply:
Suggest renaming `ManifestAnnotationSlice` to `ManifestAnnotations`

Reply:
updated